src/bt.rs

```diff
@@ -74,324 +73,217 @@ pub enum BinaryTreeError<V> {
 /// Used to indicate a traversal path on the binary tree.
 pub type Path = BitSlice;
 
-type SubTree<V> = Option<Box<Node<V>>>;
-
 /// Represents a node of a binary tree.
 pub struct Node<V> {
     value: V,
-    left: SubTree<V>,
-    right: SubTree<V>,
+    left: Option<usize>,
+    right: Option<usize>,
 }
 
-impl<V> Node<V> {
-    fn new(value: V) -> Self {
-        Self {
-            value,
-            left: None,
-            right: None,
-        }
-    }
+/// Represents an append-only binary tree.
+pub struct BinaryTree<V> {
+    nodes: Vec<Node<V>>,
+    root: Option<usize>,
```

Comment:
Is `root` necessary? We could use a convention like:
* If `nodes` is empty, the tree is empty & there is no root.
* If `nodes` is not empty, the root is at index 0.

(Indeed, I think there is no way to generate a tree whose root is not at index 0 -- the decoding code for `BinaryTree` already assumes this.)

Reply:
not really needed, thanks for pointing this out.